ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModel.kt

```diff
@@ -162,38 +160,56 @@ internal class PaywallViewModelImpl(
         }
     }
 
-    private fun updateOffering() {
+    private fun updateState() {
         viewModelScope.launch {
             try {
-                val offerings = Purchases.sharedInstance.awaitOfferings()
-                val currentOffering = options.offeringSelection.offeringIdentifier?.let { offerings[it] }
-                    ?: offerings.current
+                var currentOffering = options.offeringSelection.offering
+                if (currentOffering == null) {
+                    val offerings = Purchases.sharedInstance.awaitOfferings()
+                    options.offeringSelection.offeringIdentifier?.let { offerings[it] }
+                        ?: offerings.current
```

Comment:
Hmm I think we're not using the result from this? I think we want to save this into `currentOffering` correct?

Reply:
OMG yeah 🤦🏻  we really need tests for this class.